RedisClusterProperties throws nullpointer on empty NAT translation table

When running in cluster mode with no NAT table properties, application fails to start
`Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because "this.natTranslation" is null`

--- a/bankid-api/pom.xml
+++ b/bankid-api/pom.xml
@@ -9,7 +9,7 @@
   <parent>
     <groupId>se.swedenconnect.bankid</groupId>
     <artifactId>saml-bankid-idp-parent</artifactId>
-    <version>1.1.1</version>
+    <version>1.1.2-SNAPSHOT</version>
   </parent>
 
   <name>Sweden Connect :: BankID :: Relying Party API</name>

--- a/bankid-api/src/main/java/se/swedenconnect/bankid/rpapi/LibraryVersion.java
+++ b/bankid-api/src/main/java/se/swedenconnect/bankid/rpapi/LibraryVersion.java
@@ -24,7 +24,7 @@ public final class LibraryVersion {
 
   private static final int MAJOR = 1;
   private static final int MINOR = 1;
-  private static final int PATCH = 1;
+  private static final int PATCH = 2;
 
   /**
    * Global serialization value for classes.

--- a/bankid-frontend/pom.xml
+++ b/bankid-frontend/pom.xml
@@ -9,7 +9,7 @@
   <parent>
     <groupId>se.swedenconnect.bankid</groupId>
     <artifactId>saml-bankid-idp-parent</artifactId>
-    <version>1.1.1</version>
+    <version>1.1.2-SNAPSHOT</version>
   </parent>
 
   <name>Sweden Connect :: BankID :: SAML Identity Provider :: Frontend Distribution</name>

--- a/bankid-idp/pom.xml
+++ b/bankid-idp/pom.xml
@@ -9,7 +9,7 @@
   <parent>
     <groupId>se.swedenconnect.bankid</groupId>
     <artifactId>saml-bankid-idp-parent</artifactId>
-    <version>1.1.1</version>
+    <version>1.1.2-SNAPSHOT</version>
   </parent>
 
   <name>Sweden Connect :: BankID :: SAML Identity Provider</name>

--- a/bankid-idp/src/main/java/se/swedenconnect/bankid/idp/ApplicationVersion.java
+++ b/bankid-idp/src/main/java/se/swedenconnect/bankid/idp/ApplicationVersion.java
@@ -25,7 +25,7 @@ public final class ApplicationVersion {
 
   private static final int MAJOR = 1;
   private static final int MINOR = 1;
-  private static final int PATCH = 1;
+  private static final int PATCH = 2;
 
   /**
    * Global serialization value for classes.

--- a/bankid-idp/src/main/java/se/swedenconnect/bankid/idp/config/session/RedissonAddressCustomizers.java
+++ b/bankid-idp/src/main/java/se/swedenconnect/bankid/idp/config/session/RedissonAddressCustomizers.java
@@ -15,13 +15,13 @@
  */
 package se.swedenconnect.bankid.idp.config.session;
 
+import org.redisson.config.ClusterServersConfig;
+import org.redisson.config.SingleServerConfig;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.BiFunction;
 import java.util.function.Function;
-
-import org.redisson.config.ClusterServersConfig;
-import org.redisson.config.SingleServerConfig;
 
 /**
  * Customizers to handle a bug where the protocol section of the address becomes non-TLS when TLS is enabled.
@@ -42,7 +42,9 @@ public class RedissonAddressCustomizers {
           addresses.add(addr);
         });
         config.setNodeAddresses(addresses);
-        config.setNatMapper(clusterProperties.createNatMapper());
+        if (clusterProperties.getNatTranslation() != null) {
+          config.setNatMapper(clusterProperties.createNatMapper());
+        }
         return config;
       };
 

--- a/bankid-idp/src/test/java/se/swedenconnect/bankid/idp/ApplicationVersionTest.java
+++ b/bankid-idp/src/test/java/se/swedenconnect/bankid/idp/ApplicationVersionTest.java
@@ -15,10 +15,10 @@
  */
 package se.swedenconnect.bankid.idp;
 
-import java.util.Properties;
-
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
+
+import java.util.Properties;
 
 /**
  * Test cases for ApplicationVersion.
@@ -33,7 +33,6 @@ public class ApplicationVersionTest {
   public ApplicationVersionTest() throws Exception {
     final Properties properties = new Properties();
     properties.load(this.getClass().getClassLoader().getResourceAsStream("version.properties"));
-
     this.version = properties.getProperty("bankid.idp.version");
     if (this.version.endsWith("-SNAPSHOT")) {
       this.version = this.version.substring(0, version.length() - 9);

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
   <groupId>se.swedenconnect.bankid</groupId>
   <artifactId>saml-bankid-idp-parent</artifactId>
   <packaging>pom</packaging>
-  <version>1.1.1</version>
+  <version>1.1.2-SNAPSHOT</version>
 
   <name>Sweden Connect :: BankID :: Parent POM</name>
   <description>Parent POM for BankID SAML Identity Provider</description>

